### PR TITLE
Make run-spec-wasm2c platform-consistent

### PR DESCRIPTION
**Pull request by SoniEx2**
Or uh platform-independent? Basically cleans up the need for the harness-windows/ stuff.

### CODE REVIEW

File: test/run-spec-wasm2c.py [hunk 1]
```diff
@@ -234,7 +235,7 @@ def _MaybeWriteDummyModule(self):
             self.commands.insert(0, dummy_command)
 
     def _WriteFileAndLine(self, command):
-        self.out_file.write('// %s:%d\n' % (self.source_filename, command['line']))
+        self.out_file.write('#line {line:d} "{name:s}"\n'.format(name=self.source_filename, line=command['line']))
```
Comment (sbc100) [inline test/run-spec-wasm2c.py:238]: Maybe we could use f-string here instead.

e.g. `f'#line {command["line:]} "{self.source_filename}"'`

**Review by sbc100 — APPROVED** [review]

File: test/run-spec-wasm2c.py (hunk 1 above)
Comment (SoniEx2) [reply test/run-spec-wasm2c.py:238]: No, we still target Python 3.5 allegedly. But this did remind us to clean up the `RunWithArgsForError` stuff we added to `utils.py` now that it is no longer needed...

Comment (sbc100) [reply test/run-spec-wasm2c.py:238]: sgtm

**Comment by SoniEx2** [comment]
(any reason this hasn't been merged yet?)

**Comment by sbc100** [comment]
Looks like you need to update from main "This branch is out-of-date with the base branch". 

I guess this repo requires changes to be up-to-date with main before they land?

File: test/harness/wasm2c/floating_point.txt [hunk 2]
```diff
@@ -10,14 +9,10 @@
...
 (;; STDERR ;;;
-Error running "out/test/harness/wasm2c/floating_point/floating_point" (1):
-0/4 tests passed.
-
-out/test/harness/wasm2c/floating_point/floating_point-main.c:384: assertion failed: in w2c_floating__point__0__wasm_f32(&floating__point__0__wasm_instance, 0.f): expected 123.400002, got 0.
-out/test/harness/wasm2c/floating_point/floating_point-main.c:387: assertion failed: in w2c_floating__point__0__wasm_f32(&floating__point__0__wasm_instance, 123.400002f): expected 0, got 123.400002.
-out/test/harness/wasm2c/floating_point/floating_point-main.c:390: assertion failed: in w2c_floating__point__0__wasm_f64(&floating__point__0__wasm_instance, 0.0000000000000000): expected 123.40000000000001, got 0.
-out/test/harness/wasm2c/floating_point/floating_point-main.c:393: assertion failed: in w2c_floating__point__0__wasm_f64(&floating__point__0__wasm_instance, 123.40000000000001): expected 0, got 123.40000000000001.
-
+floating_point.txt:7: assertion failed: in w2c_floating__point__0__wasm_f32(&floating__point__0__wasm_instance, 0.f): expected 123.400002, got 0.
```
Comment (keithw) [inline test/harness/wasm2c/floating_point.txt:12]: I'm sorry for the late comment on this, but I really rely on these line numbers when debugging/developing, to find the failing assertion and the comment that leads to the line number in the original WAST file. Is there any way to keep these line numbers in our output in a cross-platform way?

Comment (keithw) [reply test/harness/wasm2c/floating_point.txt:12]: (The filename can definitely go; it's just the line number (e.g. the ":384") that I find valuable.)

Comment (SoniEx2) [reply test/harness/wasm2c/floating_point.txt:12]: oh! we mean if you use "find in file" you can search the new files for e.g. `#line 7 "floating_point.txt"` to find the relevant C line where the assert was... (so it does require a change in workflow, but hopefully not too bad?)

there's no portable way to set the name without setting the line, unfortunately, so we grabbed the most convenient/appropriate line source (the wast line).

**Pull request merged**